pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -1471,11 +1823,377 @@ def add_lift_test_measurements(
+        num_days : int
+            The number of days over which the budget is to be allocated.
+        budget_bounds : dict[str, list[Any]], optional
+            A dictionary specifying the lower and upper bounds for the budget allocation
+            for each channel. If None, no bounds are applied.
+        custom_constraints : dict[str, float], optional
+            Custom constraints for the optimization. If None, no custom constraints are applied.
+        quantile : float, optional
+            The quantile to use for recovering transformation parameters. Default is 0.5.
+
+        Returns
+        -------
+        az.InferenceData
+            The posterior predictive samples generated from the synthetic dataset.
+
+        Raises
+        ------
+        ValueError
+            If the time granularity is not supported.
+        """
+        parameters_mid = self.format_recovered_transformation_parameters(
+            quantile=quantile
+        )
+
+        scale_budget = budget / self.channel_transformer["scaler"].scale_.max()
+
+        if isinstance(budget_bounds, dict):
+            scale_budget_bounds: dict[str, tuple[float, float]] | None = {
+                k: (
+                    v[0] / self.channel_transformer["scaler"].scale_.max(),
+                    v[1] / self.channel_transformer["scaler"].scale_.max(),
+                )
+                for k, v in budget_bounds.items()
+            }
+        else:
+            scale_budget_bounds = None
+
+        allocator = BudgetOptimizer(
+            adstock=self.adstock,
+            saturation=self.saturation,
+            parameters=parameters_mid,
+            adstock_first=self.adstock_first,
+            num_days=num_days,
+        )
+
+        self.optimal_allocation_dict, _ = allocator.allocate_budget(
+            total_budget=scale_budget,
+            budget_bounds=scale_budget_bounds,
+            custom_constraints=custom_constraints,
+        )
+
+        inverse_scaled_channel_spend = self.channel_transformer.inverse_transform(
+            np.array([list(self.optimal_allocation_dict.values())])
+        )
+        original_scale_allocation_dict = {
+            k: v
+            for k, v in zip(
+                self.optimal_allocation_dict.keys(),
+                inverse_scaled_channel_spend[0],
+                strict=False,
+            )
+        }
+
+        synth_dataset = self._create_synth_dataset(
+            df=self.X,
+            date_column=self.date_column,
+            allocation_strategy=original_scale_allocation_dict,
+            channels=self.channel_columns,
+            controls=self.control_columns,
+            target_col=self.output_var,
+            time_granularity=time_granularity,
+            time_length=num_days,
+            lag=self.adstock.l_max,
+        )
+
+        synth_dataset[self.date_column] = pd.to_datetime(
+            synth_dataset[self.date_column]
+        )
```

Comment:
Can we add this casting step directly into the `_create_synth_dataset` function?